unit_test/test_crypt/CMakeLists.txt

```diff
@@ -38,6 +38,10 @@ SET(test_crypt_LIBRARY
     spdm_crypt_ext_lib
 )
 
+if(CRYPTO STREQUAL "openssl")
+    add_definitions(-DLIBSPDM_USE_OPENSSL=1)
```

Comment:
If `LIBSPDM_USE_OPENSSL` is false, there is no guarantee some crypto functions are enabled/disabled.

For consistency, I recommend:
1) we define a test_mask, for all functions, match - crypto configuration in spdm_lib_config.h
```
LIBSPDM_RSA_SSA_SUPPORT_TEST
LIBSPDM_RSA_PSS_SUPPORT_TEST
LIBSPDM_ECDSA_SUPPORT_TEST
LIBSPDM_SM2_DSA_SUPPORT_TEST
LIBSPDM_EDDSA_ED25519_SUPPORT_TEST
LIBSPDM_EDDSA_ED448_SUPPORT_TEST
LIBSPDM_FFDHE_SUPPORT_TEST
LIBSPDM_ECDHE_SUPPORT_TEST
LIBSPDM_SM2_KEY_EXCHANGE_SUPPORT_TEST
LIBSPDM_AEAD_GCM_SUPPORT_TEST
LIBSPDM_AEAD_CHACHA20_POLY1305_SUPPORT_TEST
LIBSPDM_AEAD_SM4_SUPPORT_TEST
LIBSPDM_SHA256_SUPPORT_TEST
LIBSPDM_SHA384_SUPPORT_TEST
LIBSPDM_SHA512_SUPPORT_TEST
LIBSPDM_SHA3_256_SUPPORT_TEST
LIBSPDM_SHA3_384_SUPPORT_TEST
LIBSPDM_SHA3_512_SUPPORT_TEST
LIBSPDM_SM3_256_SUPPORT_TEST
```

2) We may use `LIBSPDM_USE_OPENSSL` to configure above macro.
3) We may use `LIBSPDM_USE_MBEDTLS` to configure above macro as well.

Reply:
Thanks. I have added many macro in CMakelists.